### PR TITLE
Persist waku message timestamp & convert receiver timestamp data type to float64

### PR DESCRIPTION
**Pull request by staheri14**
This PR closes https://github.com/status-im/nim-waku/issues/596
- I also sneaked in and fixed an issue regarding the data type of receiver timestamp. This issue is that the db column holding receiver timestamp is set to `INTEGER` type, while timestamp is a `float64` value. Due to this, we would lose data precision during the insertion to  and reading from the db. The type now is changed to `float64` (the db column holds it as a seq of bytes and proper conversion takes place while reading and writing).
- The db column names are updated to better reflect the origin of the timestamps (two types exist, one for the sender and one for the receiver)

### CODE REVIEW

File: waku/v2/node/storage/message/waku_message_store.nim [hunk 1]
```diff
@@ -23,6 +23,20 @@ type
   WakuMessageStore* = ref object of MessageStore
     database*: SqliteDatabase
 
+proc toBytes(x: float64): seq[byte] =
+  let xbytes =  cast[array[0..7, byte]](x)
+  return @xbytes
+
+proc fromBytes(T: type float64, bytes: seq[byte]): T =
```
Comment (oskarth) [inline waku/v2/node/storage/message/waku_message_store.nim:30]: Hm, any reason we don't use nim-stew/byteutils for this? Seems error-prone

File: waku/v2/node/storage/message/waku_message_store.nim [hunk 2]
```diff
@@ -31,11 +45,12 @@ proc init*(T: type WakuMessageStore, db: SqliteDatabase): MessageStoreResult[T]
   let prepare = db.prepareStmt("""
     CREATE TABLE IF NOT EXISTS """ & TABLE_TITLE & """ (
         id BLOB PRIMARY KEY,
-        timestamp INTEGER NOT NULL,
+        receiverTimestamp BLOB NOT NULL,
```
Comment (oskarth) [inline waku/v2/node/storage/message/waku_message_store.nim:48]: Does this require db migrations? What does a graceful upgrade path look like here?

**Comment by jm-clius** [comment]
Agree re migration - perhaps we should update the `user_version` and upgrade all earlier versions on startup?

For interest's sake, do we not support [`REAL` type](https://www.sqlite.org/datatype3.html)?

File: waku/v2/node/storage/message/waku_message_store.nim (hunk 1 above)
Comment (staheri14) [reply waku/v2/node/storage/message/waku_message_store.nim:30]: Haven't found any support for `float64` in nim-stew/byteutils

**Comment by staheri14** [comment]
> Agree re migration - perhaps we should update the `user_version` and upgrade all earlier versions on startup?
> 
> For interest's sake, do we not support [`REAL` type](https://www.sqlite.org/datatype3.html)?

Good question,`REAL` is certainly supported by sqlite3 in general, but it is missing in the `sqlite.nim` module and inside the parameters binding proc i.e., https://github.com/status-im/nim-waku/blob/b615c6aa8bb90062ac157a49b77cd838e8d8a925/waku/v2/node/storage/sqlite.nim#L130 (perhaps it can be added, I will work on it in the subsequent PRs)
Conversion to the byte sequence is a workaround that works fine.

**Comment by staheri14** [comment]
> Agree re migration - perhaps we should update the `user_version` and upgrade all earlier versions on startup

@jm-clius 
Sure, I am reading up about migration through `user-version` pragma. In the meantime, if you have seen or done similar migration, that would be great if you could please let me know or give me some pointers, Thanks!

File: waku/v2/node/storage/message/waku_message_store.nim (hunk 2 above)
Comment (staheri14) [reply waku/v2/node/storage/message/waku_message_store.nim:48]: A short-term solution is to delete the existing db given that no important data would be lost. In the meanwhile, I am working on a decent migration approach using `user-version` pragma.

**Comment by oskarth** [comment]
Not exactly the same solution, but perhaps some inspiration re DB migrations in Nimbus:

 https://github.com/status-im/nimbus-eth2/blob/eb55fe241908b35db179aea04ae1db37a7d18b9e/beacon_chain/validators/slashing_protection_v2.nim#L628 and https://github.com/status-im/nimbus-eth2/blob/eb55fe241908b35db179aea04ae1db37a7d18b9e/tests/slashing_protection/test_migration.nim
 
 Appears they don't use sqlite built-in style, but instead did this to migrate from KV DB to sqlite.

**Review by EbubeUd — APPROVED** [review]

**Comment by D4nte** [comment]
Is this fixing https://github.com/status-im/nim-waku/issues/582 too?

**Comment by jm-clius** [comment]
If found [this](https://levlaz.org/sqlite-db-migrations-with-pragma-user_version/) blog post which I think does a fair explanation of using the `user_version` pragma when migrating. You'll have to bump `user_version` and then run the migration script for all previous `user_versions`. We'll likely build up a list of migration scripts this way (we can keep them in a separate folder/module) that will have to be run in consecutive order if someone were to upgrade from, say `user_version=1` to `user_version=10`.

**Comment by staheri14** [comment]
> Is this fixing #582 too?

@D4nte 
No, it is only about storing message timestamps in the db but not generating them.

**Comment by staheri14** [comment]
> If found [this](https://levlaz.org/sqlite-db-migrations-with-pragma-user_version/) blog post which I think does a fair explanation of using the `user_version` pragma when migrating. You'll have to bump `user_version` and then run the migration script for all previous `user_versions`. We'll likely build up a list of migration scripts this way (we can keep them in a separate folder/module) that will have to be run in consecutive order if someone were to upgrade from, say `user_version=1` to `user_version=10`.

Thanks @jm-clius, @oskarth for your direction. I am on it!

**Comment by staheri14** [comment]
For this PR, I am assuming that it  should wait for the migration script to be finalized. Rather than that, it is ready to be merged. @jm-clius @oskarth

**Comment by jm-clius** [comment]
Yes, let's wait for the migration script. Thanks!

File: waku/v2/node/storage/message/waku_message_store.nim (hunk 1 above)
Comment (oskarth) [reply waku/v2/node/storage/message/waku_message_store.nim:30]: Might want to create an upstream issue (and keep code local here for now)

**Comment by oskarth** [comment]
Sounds good to me!

**Comment by staheri14** [comment]
Please see this PR https://github.com/status-im/nim-waku/pull/607 for the db migration, it also covers all the changes in this PR with additional fixes

**Pull request closed without merging**